src/firstboot/firstboot.c

```diff
@@ -748,13 +788,35 @@ static int process_root_password(void) {
         if (lock < 0)
                 return log_error_errno(lock, "Failed to take a lock: %m");
 
-        if (arg_delete_root_password) {
+        if (arg_copy_root_shell && arg_root) {
+                struct passwd *p;
+
+                p = getpwnam("root");
+                if (!p && errno != ENOENT) {
```

Comment:
That seems iffy. getpwnam returns NULL if the name is not found, not ENOENT.

Reply:
So I changed this to something completely different in the second commit, same with the EIO comment. Should I spend the time to fix up the first commit or should I just merge them? They're related enough for me to justify merging them.